### PR TITLE
Visualising empty slots in blade systems

### PR DESCRIPTION
**Pull request by zefciu**

### CODE REVIEW

File: src/ralph_assets/tests/unit/tests_models.py [hunk 1]
```diff
@@ -314,6 +317,32 @@ def test_asset(self):
             self.assertEqual(i + 3, history.count())
 
 
+class BladeDeviceInfoFactory(DeviceInfoFactory):
```
Comment (andrzej-jankowski) [inline src/ralph_assets/tests/unit/tests_models.py]: Better place for this factory will be ``ralph_assets.tests.util`` or similar...

File: src/ralph_assets/tests/unit/tests_models.py [hunk 2]
```diff
@@ -314,6 +317,32 @@ def test_asset(self):
...
+    """Creates DeviceInfo for elements in a blade system"""
+    slot_no = factory.Iterator(it.count(1))
+
+
+class BladeAssetFactory(DCAssetFactory):
```
Comment (andrzej-jankowski) [inline src/ralph_assets/tests/unit/tests_models.py]: ditto

File: src/ralph_assets/tests/unit/tests_models.py [hunk 3]
```diff
@@ -314,6 +317,32 @@ def test_asset(self):
...
+    device_info = factory.SubFactory(BladeDeviceInfoFactory)
+
+
+def ab_iterator():
+    for i in it.count(1):
+        yield str(i) + 'A'
+        yield str(i) + 'B'
+
+
+class BladeDeviceInfoABFactory(DeviceInfoFactory):
```
Comment (andrzej-jankowski) [inline src/ralph_assets/tests/unit/tests_models.py]: ditto

File: src/ralph_assets/tests/unit/tests_models.py [hunk 4]
```diff
@@ -314,6 +317,32 @@ def test_asset(self):
...
+    """Creates DeviceInfo for elements in a blade system (AB mode)"""
+    slot_no = factory.Iterator(ab_iterator())
+
+
+class BladeAssetABFactory(DCAssetFactory):
```
Comment (andrzej-jankowski) [inline src/ralph_assets/tests/unit/tests_models.py]: ditto

File: src/ralph_assets/tests/unit/tests_models.py (hunk 1 above)
Comment (zefciu) [inline src/ralph_assets/tests/unit/tests_models.py]: These are very specific factories needed only here.

**Comment by andrzej-jankowski** [comment]
:ok:

**Comment by ar4s** [comment]
:ok:

**Pull request merged**